packages/dev/core/src/Meshes/Node/Blocks/mappingBlock.ts

```diff
@@ -0,0 +1,143 @@
+import { NodeGeometryBlock } from "../nodeGeometryBlock";
+import type { NodeGeometryConnectionPoint } from "../nodeGeometryBlockConnectionPoint";
+import { RegisterClass } from "../../../Misc/typeStore";
+import { NodeGeometryBlockConnectionPointTypes } from "../Enums/nodeGeometryConnectionPointTypes";
+import type { NodeGeometryBuildState } from "../nodeGeometryBuildState";
+import { PropertyTypeForEdition, editableInPropertyPage } from "../../../Decorators/nodeDecorator";
+import { Vector2, Vector3 } from "../../../Maths/math.vector";
+
+/**
+ * Type of mappings supported by the mapping block
+ */
+export enum MappingTypes {
+    /** Spherical */
+    Spherical,
+    /** Cylindrical */
+    Cylindrical,
+}
+
+/**
+ * Block used to generate UV coordinates
+ */
+export class MappingBlock extends NodeGeometryBlock {
+    /**
+     * Gets or sets the mapping type used by the block
+     */
+    @editableInPropertyPage("Mapping", PropertyTypeForEdition.List, "ADVANCED", {
+        notifiers: { rebuild: true },
+        options: [
+            { label: "Spherical", value: MappingTypes.Spherical },
+            { label: "Cylindrical", value: MappingTypes.Cylindrical },
+        ],
+    })
+    public mapping = MappingTypes.Spherical;

```

Comment:
I wonder if there is some value to have the mapping being an input instead of a static parameter?

As we can build any kind of geometry with the NGE, sometimes the final mesh could be more extended in one direction than in another, so sometimes it could be better to use a spherical mapping (when the extents are roughly the same) or a cylindrical mapping (when one extent is larger than the other two).

Or, maybe have a "Auto select best mapping" choice, that would choose the Spherical or Cylindrical calculation based on the bounding box?

Reply:
good idea. I will see how to add it in a future revision. That will probably force the input to be integer but that is probably acceptable

For now I first need to add a cube projection